src/objects/zcl_abapgit_object_sfpf.clas.abap

```diff
@@ -113,6 +113,9 @@ CLASS ZCL_ABAPGIT_OBJECT_SFPF IMPLEMENTATION.
     TRY.
         li_wb_form = load( ).
         li_fp_form ?= li_wb_form->get_object( ).
+
+        mo_files->add_raw( iv_ext = 'xdp' iv_data = li_fp_form->get_layout( )->get_layout_data( ) ).
```

Comment:
this adds the data in a new file, but should it also be cleared somewhere in the existing XML? Plus consider how to make this downwards compatible

Reply:
> this adds the data in a new file, but should it also be cleared somewhere in the existing XML?

The previous version had this statement:
`li_fp_form->get_layout( )->set_layout_data( i_layout_data = conv #('') ).`

But I removed it because it did not save any space in the SFPF XML file (because there
are layouts for other languages) and the statement instantly removes
the layout from caches and breaks forms.

>  Plus consider how to make this downwards compatible

Yay, pull without the corresponding XDP file will remove the layout!

Reply:
@larshp The backward compatibility. I would add a check that *xdp* file is in *mo_files* and if not present, I would skip:
`li_form->get_layout( )->set_layout_data( i_layout_data = mo_files->read_raw( iv_ext = 'xdp' ) ).`

The question is how to implement the check. In other languages, I would just try to call `read_raw` and catch `file not found` exception. If this approach is ok to you, I need to add a new exception type and change the type of raised exception in the class *zcl_abapgit_objects_files*.

The other more complicated way is to search the file file in the result of `get_files()`.

The last option is to add a helper method testing presence to the class *zcl_abapgit_objects_files*.

What's your preference?

Reply:
suggest adding a new method in `ZCL_ABAPGIT_OBJECTS_FILES` named `file_exists` which returns a boolean, this describes what the code does, plus it avoids having exceptions which are slow